src/coreclr/jit/emitloongarch64.cpp

```diff
@@ -5029,88 +5433,87 @@ void emitter::emitDisInsName(code_t code, const BYTE* addr, instrDesc* id)
             printf("fmadd.s      %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FMADD_D:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fmadd.d      %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FMSUB_S:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fmsub.s      %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FMSUB_D:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fmsub.d      %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FNMADD_S:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fnmadd.s     %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FNMADD_D:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fnmadd.d     %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FNMSUB_S:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fnmsub.s     %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
         case LA_4R_FNMSUB_D:
-        {
-            const char* fd = RegNames[(code & 0x1f) + 32];
-            const char* fj = RegNames[((code >> 5) & 0x1f) + 32];
-            const char* fk = RegNames[((code >> 10) & 0x1f) + 32];
-            const char* fa = RegNames[((code >> 15) & 0x1f) + 32];
             printf("fnmsub.d     %s, %s, %s, %s\n", fd, fj, fk, fa);
             return;
-        }
+
+#ifdef FEATURE_SIMD
+        case 0x91:
```

Comment:
Aren't there `#define` values / names for these numbers?

I'm surprised the disassembly output has all these hard-coded text names instead of having the text come from the instruction table. If they came from a table, then all this code would reduce to a couple lines, since they have the same set of arguments (for `v` and `xv` cases).

Reply:
I guess this comment applies to this entire disassembler, which seems insufficiently data driven. So it's nothing new.

There are an extraordinary number of "magic numbers" in the code.

Reply:
It seems like this whole thing could be written much better, more data driven, and not require an additional 10,000 lines of code.

Reply:
Thanks.
As the ISA's encoding is very stable and only used for debugging, only pushed one time, here using the explicit ‘‘magic numbers’’.
If needed, we can amend these further.

Reply:
@BruceForstall 
I will reduce the lines for this disasm function by pushing a patch later.

Reply:
Hi, @BruceForstall 

The previous LoongArch64's implementation of the `emitDisInsName()` is so large as we disassemble from the memory and decoding one by one fields.

For ARM64 and X86, issuing the instruction from the `instrDesc` to the memory, while the disassembly is also from the `instrDesc` other than from the memory.
BTW, I think disassembling from the memory is better.

I have redesigned the `emitDisInsName()`.
Thanks very much for your reviewing and advices!!!